File: playground-golang/examples/n1/n1.go
package n1

func twoSum(nums []int, target int) []int {
	dict := make(map[int]int, len(nums))

	for i, n := range nums {
		if idx, ok := dict[n]; ok {
			return []int{idx, i}
		}
		dict[target-n] = i
	}

	panic("")
}

File: playground-typescript/src/n1/n1.ts
export function twoSum(nums: number[], target: number): number[] {
    const dict = new Map<number, number>();
    for (let i = 0; i < nums.length; ++i) {
        const remainIdx = dict.get(nums[i]);
        if (typeof remainIdx === 'number')
            return [remainIdx, i];
        else
            dict.set(target - nums[i], i);
    }

    throw new Error();
};